File: guessing-game/guessing_game.py
import random
print("Welcome to this guessing game. Enter a number from 1-10: ")
correct_guess = False

while(correct_guess == False):
    try:
        num = int(input())
    except:
        print("You entered an invalid value.")
        break
    randnum = random.randrange(10)

    if (num == randnum):
        print(f"You have guessed correctly. The number is {randnum}\n")
        correct_guess=True
    else:
        print(f"You have guessed incorrectly. Try again.")

File: guessing-game/guessing_game.java
import java.util.Random;
import java.util.Scanner;

public class guessing_game {
    public static void main(String[] args) {

        boolean correctGuess = false;

        System.out.println("Enter a number from 1-10: ");

        Scanner sc = new Scanner(System.in);

        Random rnd = new Random();

        int num = rnd.nextInt(10) + 1;

        while (correctGuess == false) {
            int guess = sc.nextInt();
        
            if (guess == num) {
                System.out.println("You guessed correct! The number is " + num);
                break;
            }

            else {
                System.out.println("Incorrect guess! Try again.");
            }
        }

        sc.close();
    }

}